### ISSUE TITLE
Nord VPN Auth Issue

### ISSUE THREAD

**Issue by Rednjak** [haugene/docker-transmission-openvpn#1092]
**Describe the problem**
I am using the latest ARM image on the raspberry pi as it is shown in the example file `docker-compose-armhf.yml`, but without the `proxy` and `rss` part.

```
  
version: '2'

services:
 transmission:
  image: haugene/transmission-openvpn:latest-armhf
  cap_add:
    - NET_ADMIN
  devices:
    - "/dev/net/tun"
  restart: always
  ports:
    - "9091:9091"
    - "8888:8888"
  dns:
    - 8.8.8.8
    - 8.8.4.4
  volumes:
    - /etc/localtime:/etc/localtime:ro
    - /home/Shared/:/data
  environment:
    - OPENVPN_PROVIDER=NORDVPN
    - OPENVPN_USERNAME=myemail+nord@gmail.com
    - OPENVPN_PASSWORD=password
    - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
    - LOCAL_NETWORK=192.168.0.0/24
```

**Logs**
```
Creating network "transmission_default" with the default driver
Creating transmission_transmission_1 ... done
Attaching to transmission_transmission_1
transmission_1  | Using OpenVPN provider: NORDVPN
transmission_1  | 2020-03-24 21:04:21 Checking curl installation
transmission_1  | 2020-03-24 21:04:21 Removing existing configs
transmission_1  | 2020-03-24 21:04:21 Selecting the best server...
transmission_1  | 2020-03-24 21:04:21 Searching for group: legacy_p2p
transmission_1  | 2020-03-24 21:04:21 Searching for technology: openvpn_udp
transmission_1  | 2020-03-24 21:04:21 Best server : at53.nordvpn.com
transmission_1  | 2020-03-24 21:04:21 Downloading config: default.ovpn
transmission_1  | 2020-03-24 21:04:21 Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_udp/servers/at53.nordvpn.com.udp.ovpn
transmission_1  |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
transmission_1  |                                  Dload  Upload   Total   Spent    Left  Speed
100  2810  100  2810    0     0   4390      0 --:--:-- --:--:-- --:--:--  4383
transmission_1  | 2020-03-24 21:04:21 Checking line endings
transmission_1  | 2020-03-24 21:04:21 Updating configs for docker-transmission-openvpn
transmission_1  | Starting OpenVPN using config default.ovpn
transmission_1  | Setting OPENVPN credentials...
transmission_1  | adding route to local network 192.168.0.0/24 via 192.168.16.1 dev eth0
transmission_1  | Tue Mar 24 21:04:22 2020 OpenVPN 2.4.7 arm-unknown-linux-gnueabihf [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Feb 20 2019
transmission_1  | Tue Mar 24 21:04:22 2020 library versions: OpenSSL 1.1.1d  10 Sep 2019, LZO 2.10
transmission_1  | Tue Mar 24 21:04:22 2020 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
transmission_1  | Tue Mar 24 21:04:22 2020 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
transmission_1  | Tue Mar 24 21:04:22 2020 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication
transmission_1  | Tue Mar 24 21:04:22 2020 TCP/UDP: Preserving recently used remote address: [AF_INET]185.216.34.221:1194
transmission_1  | Tue Mar 24 21:04:22 2020 Socket Buffers: R=[163840->163840] S=[163840->163840]
transmission_1  | Tue Mar 24 21:04:22 2020 UDP link local: (not bound)
transmission_1  | Tue Mar 24 21:04:22 2020 UDP link remote: [AF_INET]185.216.34.221:1194
transmission_1  | Tue Mar 24 21:04:22 2020 TLS: Initial packet from [AF_INET]185.216.34.221:1194, sid=d78ce554 fc341ed8
transmission_1  | Tue Mar 24 21:04:22 2020 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this
transmission_1  | Tue Mar 24 21:04:22 2020 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA
transmission_1  | Tue Mar 24 21:04:23 2020 VERIFY OK: depth=1, C=PA, O=NordVPN, CN=NordVPN CA4
transmission_1  | Tue Mar 24 21:04:23 2020 VERIFY KU OK
transmission_1  | Tue Mar 24 21:04:23 2020 Validating certificate extended key usage
transmission_1  | Tue Mar 24 21:04:23 2020 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication
transmission_1  | Tue Mar 24 21:04:23 2020 VERIFY EKU OK
transmission_1  | Tue Mar 24 21:04:23 2020 VERIFY OK: depth=0, CN=at53.nordvpn.com
transmission_1  | Tue Mar 24 21:04:25 2020 Control Channel: TLSv1.2, cipher TLSv1.2 ECDHE-RSA-AES256-GCM-SHA384, 4096 bit RSA
transmission_1  | Tue Mar 24 21:04:25 2020 [at53.nordvpn.com] Peer Connection Initiated with [AF_INET]185.216.34.221:1194
transmission_1  | Tue Mar 24 21:04:26 2020 SENT CONTROL [at53.nordvpn.com]: 'PUSH_REQUEST' (status=1)
transmission_1  | Tue Mar 24 21:04:26 2020 AUTH: Received control message: AUTH_FAILED
transmission_1  | Tue Mar 24 21:04:26 2020 SIGTERM[soft,auth-failure] received, process exiting
```

**Host system:**
Raspberry Pi
Docker 19.03.08
docker-compose 1.25.4 

I tried to wrap the environment variables in `"` and `'`, but it didn't help. I also change my password to not contain any special character and it also didn't help. The only thing that comes to my mind is `+` in my email, but I am not totally convinced by it. I also tried loading the credentials in a file and `docker run` command and both resulted in `AUTH_FAILED`

**Comment by edgd1er** [comment]
hi,

according to [yamls specs](https://yaml.org/spec/1.2/spec.html#id2795688), + is special sign, you may try to escape with a \ or try to encode : +002B

`
OPENVPN_USERNAME=myemail\+nord@gmail.com
`

or

`
OPENVPN_USERNAME=myemail\u002Bnord@gmail.com
`
all this considering that your login/password are corrects ;)

**Comment by haugene** [comment]
And if that doesn't work you can also mount a file to "/config/openvpn-credentials.txt" that contains your username on the first line and your password on the second. If you provide this you don't have to set OPENVPN_USERNAME and OPENVPN_PASSWORD.

This is an old issue #245

**Comment by Rednjak** [comment]
@edgd1er Thank you for the suggestion. I tried both and it did not work.
@haugene I already tried this method and it did not help.

I solved my problem by contacting NordVPN support to change my email address without the `+`. As soon as they confirmed the changed I ran the `docker-compose` and it spanned up :D 

It would be interesting to investigate this issue though. I think I am not the only one who is making use of the `+` in the email to track down spam :D 

I will go ahead and close this issue. @edgd1er @haugene  thank you again for the fast responses
